core/transcoder.go

```diff
@@ -122,3 +107,40 @@ func parseURI(uri string) (string, uint64, error) {
 	seqNo, err := strconv.ParseUint(parts[0], 10, 64)
 	return mid, seqNo, err
 }
+
+func resToTranscodeData(res *ffmpeg.TranscodeResults, opts []ffmpeg.TranscodeOptions) (*TranscodeData, error) {
+	if len(res.Encoded) != len(opts) {
+		return nil, errors.New("lengths of results and options different")
+	}
+
+	// Convert results into in-memory bytes following the expected API
+	segments := make([]*TranscodedSegmentData, len(opts), len(opts))
+	for i := range opts {
+		oname := opts[i].Oname
+		o, err := ioutil.ReadFile(oname)
+		if err != nil {
+			glog.Error("Cannot read transcoded output for ", oname)
+			return nil, err
+		}
+		segments[i] = &TranscodedSegmentData{Data: o, Pixels: res.Encoded[i].Pixels}
+		os.Remove(oname)
+	}
+
+	return &TranscodeData{
+		Segments: segments,
+		Pixels:   res.Decoded.Pixels,
+	}, nil
+}
+
+func profilesToTranscodeOptions(workDir string, accel ffmpeg.Acceleration, profiles []ffmpeg.VideoProfile) []ffmpeg.TranscodeOptions {
+	opts := make([]ffmpeg.TranscodeOptions, len(profiles), len(profiles))
+	for i := range profiles {
+		o := ffmpeg.TranscodeOptions{
+			Oname:   fmt.Sprintf("%s/out_%s.ts", workDir, common.RandName()),
```

Comment:
Since both the local and nvidia transcoders use this helper to transform video profiles into a slice of transcode options, both transcoders will use random names for output files in their configured `workDir`.

Previously, the local transcoder created a [directory with a random name](https://github.com/livepeer/go-livepeer/blob/master/core/transcoder.go#L30) in its `workDir` and then *did not* use random names for output files in the directory. Meanwhile, the nvidia transcoder used random names for output files in its configured `workDir`.